### ISSUE TITLE
double-clicking address or name in PIE binary seeks to the wrong address

### ISSUE THREAD

**Issue by iGr33k** [rizinorg/rizin#1828]
**Environment information**
* Operating System: ALL
* Cutter version: 2.0.3
* Obtained from:
  - Downloaded release from Cutter website or GitHub 
* File format: ELF

**Describe the bug**

**To Reproduce**

1. Open Binary
2. Double-Click on the name or address

**Expected behavior**

Open PIE binary and click on address analyzed as `main`, in this example it's `0x433c`, Cutter seeks to `0x3214` instead.

**Screenshots**

**Additional context**
Example binary is in the attachments.
[ziy.zip](https://github.com/rizinorg/cutter/files/7293778/ziy.zip)

**Comment by DMaroo** [comment]
If you see Cutter's output on the console, you get the following message from Cutter when you double-click:

```
"More than one (2) references here. Weird behaviour expected."
```

As the message explains, Cutter found more than two addresses referencing the `main` symbol. In such a case, the first xref is chosen which, unfortunately, is not the correct address of `main`. The real problem is why are there 2 xrefs.  This is a Rizin problem and not a Cutter problem. The above can be reproduced in Rizin by seeking to `0x1121` (where `main` is referenced), and then using `axf`, which will output two offsets, one of which is bogus.

@ret2libc @thestr4ng3r I suggest this issue be moved to Rizin.

**Comment by ret2libc** [comment]
By the way, I think the fact that cutter uses analysis references to seek to things when double clicking, is an issue when you disable analysis, because you can't navigate anywhere.

**Comment by ret2libc** [comment]
```
[0x00001112]> pd 1 @ 0x1121
│           0x00001121      lea   rdi, [rip + 0x3214]                  ; main
│                                                                      ; 0x433c ; void *main
[0x00001112]> axf @ 0x1121
d 0x3214 nop
d 0x433c main
```

@GustavoLCR / @kazarmy do you know where to look at to check why that nop is put in the xrefs?

**Comment by DMaroo** [comment]
> By the way, I think the fact that cutter uses analysis references to seek to things when double clicking, is an issue when you disable analysis, because you can't navigate anywhere.

Yes, I agree. Jumping to memory addresses (like `0x1234`) should not require any analysis.

**Comment by iGr33k** [comment]
Issue is still present.

**Comment by karliss** [comment]
~~Can't repeat with latest rizin anymore.~~ It was still happening using few weeks old one. Was I really so lucky for it to be fixed  within last few weeks after being a problem for years?

**Comment by karliss** [comment]
Still there, extra xref is added by something in  `aaa`. Running only `aa` produces the expected result with single xref.

**Comment by karliss** [comment]
The bad xref is set by https://github.com/rizinorg/rizin/blob/d7dec467fe484acb3465ef39f5c76409b4edfbf6/librz/core/canalysis.c#L2260C20-L2263C14 

Few lines higher correct one is set by logic which is based on op.ptr, but the correct one also gets set by different analysis pass.

Like suggested probably should just try disabling it and see which test break. There are probably instructions where using disp that way makes some sense. But in general just picking an offset without more detailed checks seems like recipe for false positives. Maybe doing it only as fallback when no other xrefs from instruction have been found could reduce false positives, without too much platform specific analysis of whether it's a potential xref.

I can think of potential cases of addressing using some kind of offset:
* rip relative offset -> the problematic case in this issue 
* stack relative offset -> most common cases should be filtered by >512 check, but there might still be false positives in huge functions 
* more or less absolute value (MCU and maybe some non PIE code) -> xref should be already handled by other cases
* module relative addressing using a base offset stored in some register -> probably the case which this code tries to handle
* offsets into structures or vtables -> most of the time should be <512 
* use of `lea` for general math  or array related calculations

Will try running tests with it disabled to see which cases breaks.

**Comment by karliss** [comment]
One of the testcases relying on this codepath references https://github.com/radareorg/radare2/issues/15315 

It mentions a couple of other cases I didn't think of (today). Funny that I am partially at fault for encouraging those changes.

Total 2 broken tests:

"db/analysis/x86_32 Fix #15315 - Honor RzAnalysisOp.disp in `aar` and `ao`" bins/pe/honor_ranal_op_disp.exe 
```diff
-(nofunc) 0x4010d1 [DATA] lea edx, [ebx*4 + str.Y_B]
```
Supposedly the main testcase related to changes mentioned above.

`db/cmd/cmd_afl aflj` bins/elf/ls 
This test prints all the xrefs in an executable. Will have to look into details whether those are false positives or good xrefs.

**Comment by karliss** [comment]
Second one in bins/elf/ls is same false positive as discussed in this issue.
